### ISSUE TITLE
Story: Build a NCIDS Header Block in Drupal

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#3268]
## As a Drupal Site Builder, I want to be able to add a NCIDS Header to my Drupal site as a configuration block, so that I can provide my users a way to navigate the site following the NCIDS without having to enter complicated HTML

> **ESTIMATE** 5

### Requirements:
* Set up new NCIDS Header Block
  * Additional fields:
    * Home Page Title - aria label used in place of the alt tag for the logo
    * Search Results Page - relative URL to use as the search results page
    * Autosuggest Collection: freeform text field for search autosuggest collection
      * NOTE: If there is no autosuggest, leave this empty, but also do not wire up the autosuggest in the search box component 
* We would like the logo to be managed within the Theme settings and not requiring a release. We also need to support alternate languages and other text in the image as settings as well.
   * Use page langcode as SVG indicator thingy (#)
   * Template will have to output the langcode
   * Figure out how to allow SVGs in Drupal
* Analytics for banner, toggle, search box, and primary nav

### Notes:
* There should be a [cache-context added to the cache information](https://www.drupal.org/docs/drupal-apis/cache-api/cache-contexts#core-contexts) on the block that will vary it based on the nav that is being shown.
   * IDK what this might be at this point as the contexts are from the request standpoint, NOT from the page content standpoint.
   * Basically http://www.devbox/ and http://www.devbox/api-test should show different menus on your machine and on ODEs
* This will render the static elements of the `nci-extended` header:
   * The logo, and any alternate text
   * The primary navigation
       * If says "Hide in Main Nav", do not show - default to checked
     * Navigation label over section label
     * If the primary nav item does not have a megamenu, don't output `data-menu-id` and make it a link to the page
   * The search box (with autosuggest configured)
      * This requires the following information:
         * The URL of the search results page
         * The API base url. This should be available via Javascript as `window.CDEConfig.sitewideSearchConfig.searchApiServer`
            * So however you initialize that autosuggest instance, use that! 
               * The text input for the search box should have a `data-autosuggest-collection` attribute that contains the configured collection. 
            * And no, you should not be asking "how does Drupal read the value from the Javascript"
   * Any other bits and pieces. (like the mobile menu button)

This will NOT include any of the Javascript elements defined in the tickets below. They should enhance the page on load and can be handled separately.
* NCIOCPL/cgov-digital-platform#3277
* NCIOCPL/cgov-digital-platform#3278
* NCIOCPL/cgov-digital-platform#3279

### Acceptance Criteria

```gherkin
Scenario: A site admin sets the logo and alternate text in the Drupal ???
  Given User is a Site Admin
    And [some more context]
  When  Drupal renders the header
  Then  The logo appears as a link with the "correct accessibility information"
    And [another outcome]
```

* Site admin adds a ncids header sets "The accessible label for the home page link."
  * header renders with the following
     ```
     <a href="/" aria-label="<THE LABEL>">
        <svg role="img" width="385" height="38" aria-hidden="true">
          <use href="/profiles/custom/cgov_site/themes/custom/ncids_trans/logo.svg#en"></use>
        </svg>
      </a>
     ```
* Site admin adds a ncids header without "Search Results Page"
   * No search box appears
* Site admin selects a page from the system for the "Search Results Page"
   * when the user submits the search form they are taken to the search results page with their keyword
* Site admin sets the "The collection name for the autosuggest API."
   * the input has a data-autosuggest-collection attribute with that value
* Site admin goes to /admin/appearance/settings/ncids_trans and expands the LOGO IMAGE container. Unchecks the checkbox. Clicks choose file and uploads a different logo.
   * The new logo appears correctly for English and Spanish
* Site admin edits a Cgov Site Section and checks the "Set Main Nav Root" checkbox
   * All the immediate child Cgov Site Sections will display in the primary navigation if the following conditions are met:
      * The child site section does not have "Hide in main nav" checked under the "Navigation Display Options" heading
      * The child section has a landing page
      * The landing page has been published 
   * 
 
---

### Prerequisites
- [ ] https://github.com/NCIOCPL/ncids/wiki/Technical-NCI-Header-Design

**Comment by bennettcc** [comment]
This ticket is currently on hold - dependent upon https://github.com/NCIOCPL/cgov-digital-platform/issues/3436 to be resolved first

**Comment by bennettcc** [comment]
3 SP accomplished in S34, 2 SP to carry into S35

**Comment by sarinapadilla** [comment]
Closing this ticket as it has been fixed by PR #3449!
